### ISSUE TITLE
DKIM signing not working

### ISSUE THREAD

**Issue by Kenchava** [sympa-community/sympa#1036]
Version
-------

6.2.24

Installation method
-------------------

apt (Mail:DKIM is installed)

Expected behavior
-----------------

DKIM headers of received emails from lists should be populated.

Actual behavior
---------------

Message is not signed.

Additional information
----------------------

/etc/sympa/sympa/sympa.conf:
dkim_feature off
dkim_add_signature_to   robot,list
dkim_private_key_path /etc/sympa/sympa/dkim.private
dkim_signature_apply_on md5_authenticated_messages,smime_authenticated_messages,dkim_authenticated_messages,editor_validated_messages
dkim_signer_domain <domain name>
dkim_selector sympalists

The TXT record containing the public key is deployed at sympalists._domainkey.<domain name>

The DKIM feature is enabled on my test list as I cannot enable it globally for now.

/var/lib/sympa/list_data/<listname>/config:
dkim_feature on

**Comment by ikedas** [comment]
Hi @Kenchava ,
Could you please show us these things?
- the log of Sympa _from when_ you posted a test message _to when_ the message was delivered, and
- full header of delivered message.

**Comment by Kenchava** [comment]
Hi ikedas,

Late last evening I was actually able to get everything working. DKIM signing began working once it was enabled globally in the sympa.conf file. Originally I had dkim_feature off within sympa.conf and dkim_feature on specified on a specific list.

**Comment by ikedas** [comment]
Hi @Kenchava ,

> Hi @ikedas ,
> 
> Late last evening I was actually able to get everything working. DKIM signing began working once it was enabled globally in the sympa.conf file. Originally I had dkim_feature off within sympa.conf and dkim_feature on specified on a specific list.

Hmm, that behavior doesn't make sense.  I'll submit a pr to fix it in a few days.  Thanks for reporting.

**Comment by racke** [comment]
Looks like the following Debian bug report is also related to this problem: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=954475

**Comment by ikedas** [comment]
@racke, I think that report on Debian looks discussiong on another problem.

**Comment by ikedas** [comment]
A fix to this bug will be included in the next beta (6.2.59b.2 around 7 Dec).  It will be very appreciated to confirm this fix.

**Comment by ikedas** [comment]
Previous pr #1050 was misleading.

I checked the code from when [initially DKIM integration was added](https://github.com/sympa-community/historic-sympa/commit/52b1f0dccdddc9c89f3bc027bae487aed50a7670), and found that:
  * `dkim_feature` parameter is available only in domain context (`robot.conf` or `sympa.conf` for the default).
  * Configuration schema also defines it in list context (`config`) where it is never used.

By a new pr I'll roll back previous change and remove `dkim_feature` parameter from list context.
